texthero/_types.py

```diff
@@ -185,45 +190,50 @@ def is_numeric(x):
                 return True
 
         def is_list_of_numbers(cell):
-            return all(is_numeric(x) for x in cell) and isinstance(cell, (list, tuple))
+            return isinstance(cell, (list, tuple)) and all(is_numeric(x) for x in cell)
 
         try:
             first_non_nan_value = s.loc[s.first_valid_index()]
-            if not is_list_of_numbers(first_non_nan_value) or s.index.nlevels != 1:
-                raise TypeError(error_string)
+            if not is_list_of_numbers(first_non_nan_value):
+                return False, error_string
         except KeyError:  # Only NaNs in Series -> same warning applies
-            raise TypeError(error_string)
+            return False, error_string
+
+        return True, ""
 
 
-class RepresentationSeries(HeroSeries):
+class DataFrame(HeroTypes):
```

Comment:
Is this class really necessary?

Reply:
Yes, it makes checking the types much easier for us through `@InputSeries` and gives users a good error message / docstring. We _could_ also overwrite the Pandas DataFrame class to add the function `check_type`, but that's at least as much code and annoying for other developers. This implementation makes it easy to use with the decorator.

Reply:
Ok; I see. Sure, let's keep as it is, maybe just adding a sentence explaining why (the same answer you gave it to me there, it might be helpful for other developers too!)

Reply:
Yes, I added an explanation to the `_types` file docstring :kangaroo: :ok_hand: